src/app/public/modules/dock/dock.component.ts

```diff
@@ -100,6 +104,18 @@ export class SkyDockComponent implements OnInit {
     this.itemRefs.splice(this.itemRefs.indexOf(found), 1);
   }
 
+  public setOptions(options: SkyDockOptions): void {
```

Comment:
It seems like this should be added in a way that does not break previous implementations, right? Could this still use `ngOnInit` with default options?

Reply:
This is an odd case of us having a component that is purely internal. The only way that this component can be created by a consumer is through the dock service. We internally create the component from the service and my changes do this in a way that does not break previous implementations if no options are given.

Reply:
It's in the public API but I see what you're saying.

Reply:
Ya. It is technically public in terms of Javascript access. However, we don't export the component in the `public_api.ts` file or from the module so it isn't usable in templates and isn't exported so we have tended to consider that not a part of the "public api" in practice.